sdk/src/Services/DynamoDBv2/Custom/DocumentModel/TableBuilder.cs

```diff
@@ -0,0 +1,260 @@
+            }
+
+            return this;
+        }
+
+        /// <inheritdoc/>
+        public ITableBuilder AddLocalSecondaryIndex(string indexName, string rangeKeyAttribute, DynamoDBEntryType type)
+        {
+            if (string.IsNullOrEmpty(indexName))
+            {
+                throw new ArgumentNullException(nameof(indexName), "The name of the local secondary index is required");
+            }
+
+            if (_table.LocalSecondaryIndexes.ContainsKey(indexName))
+            {
+                throw new ArgumentException($"An local secondary index with name {indexName} has already been defined.");
+            }
+
+            if (string.IsNullOrEmpty(rangeKeyAttribute))
+            {
+                throw new ArgumentNullException(nameof(rangeKeyAttribute), "The attribute name of the range key within the local secondary index is required.");
+            }
+
+            if (_table.HashKeys.Count == 0)
+            {
+                throw new ArgumentException("A local secondary index uses the table's partition key, which was not provided. Call AddHashKey prior to AddLocalSecondaryIndex.");
+
+            }            
+
+            _table.LocalSecondaryIndexNames.Add(indexName);
+
+            _table.LocalSecondaryIndexes.Add(indexName, new LocalSecondaryIndexDescription
+            {
+                IndexName = indexName,
+                KeySchema = new List<KeySchemaElement>()
+                {
+                    new KeySchemaElement { AttributeName = _table.HashKeys[0], KeyType = KeyType.HASH },
+                    new KeySchemaElement { AttributeName = rangeKeyAttribute, KeyType = KeyType.RANGE }
+                }
+            });
+
+            if (!_attributesAlreadyProcessed.Contains(rangeKeyAttribute))
+            {
+                _table.Attributes.Add(new AttributeDefinition(rangeKeyAttribute, dynamoDBEntryTypeToScalarAttributeType(type)));
+                _attributesAlreadyProcessed.Add(rangeKeyAttribute);
+            }
+
+            return this;
+        }
+
+        /// <inheritdoc/>
+        public ITableBuilder AddGlobalSecondaryIndex(string indexName, string hashkeyAttribute, DynamoDBEntryType hashKeyType, string rangeKeyAttribute, DynamoDBEntryType rangeKeyType)
+        {
+            if (string.IsNullOrEmpty(indexName))
+            {
+                throw new ArgumentNullException(nameof(indexName), "The name of the global secondary index is required");
+            }
+
+            if (_table.GlobalSecondaryIndexes.ContainsKey(indexName))
+            {
+                throw new ArgumentException($"An global secondary index with name {indexName} has already been defined.");
+            }
+
+            if (string.IsNullOrEmpty(hashkeyAttribute))
+            {
+                throw new ArgumentNullException(nameof(hashkeyAttribute), "The attribute name of the partition key within the global secondary index is required.");
```

Comment:
Be consistent and use hash key.

Reply:
Fixed in https://github.com/aws/aws-sdk-net/compare/74020f5167a3bf2e54dd53bdeba482f965e43dd2..abcc18e865203cd0aa124c2c175f0dc798d5be93